### PR TITLE
[toggl-track] Hide archived projects

### LINKED ISSUE #7316: [Toggl Track] also shows archived projects, which is not what you would expect
### Extension

https://www.raycast.com/franzwilhelm/toggl-track

### Description

When searching for a project to track time, the results also contains projects that are archived within toggl. Which gives errors like pulling up a report and the tracked time isn't there (because it has been tracked against an archived project.
Solution: by default the extension should only return active projects. Maybe you could add an option to include archived projects as well (but I would not be using that :-) )

### Who will benefit from this feature?

Everyone, now you can make serious mistakes and end up not charging for multiple tracked hours.

### Anything else?

ONo

### PR DESCRIPTION
**Pull request by bkeys818**
## Description

Adds a new preference to the toggle-track extension that allows archived projects to be filtered out.
The default value is true.

Closes #7316

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by michaelfaisst** [comment]
Lgtm 👍

File: extensions/toggl-track/src/toggl/index.ts [hunk 1]
```diff
@@ -74,9 +76,9 @@ const TogglAPI = function (apiToken: string) {
 
 interface Preferences {
   togglApiToken: string;
+  hideArchivedProjects: boolean;
 }
 
-const preferences: Preferences = getPreferenceValues();
-const toggl = TogglAPI(preferences.togglApiToken);
+const toggl = TogglAPI();
```
Comment (franzwilhelm) [inline extensions/toggl-track/src/toggl/index.ts:82]: I suggest taking in { togglApiToken, hideArchivedProjects } as params to TogglAPI as before, now config is a side-effect within TogglAPI

Comment (franzwilhelm) [reply extensions/toggl-track/src/toggl/index.ts:82]: Other than that great work!

**Review by franzwilhelm — CHANGES_REQUESTED** [review]

File: extensions/toggl-track/src/toggl/index.ts (hunk 1 above)
Comment (bkeys818) [reply extensions/toggl-track/src/toggl/index.ts:82]: I'm cool with that. Is there any specific benefit, or is that just good practice?

**Comment by bkeys818** [comment]
Tried to do it on my phone🙄

**Comment by bkeys818** [comment]
I'd recommend doing a squash and merge.

**Review by thomaslombart — COMMENTED** [review]
Could you add an entry in the `CHANGELOG.md` file? Here's our guidelines about that: [Version History](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history).

**Review by franzwilhelm — APPROVED** [review]
✅

**Comment by thomaslombart** [comment]
Don't forget to run Prettier 😊

**Comment by bkeys818** [comment]
Found it. I'm still using v2.

**Comment by thomaslombart** [comment]
Ready to merge?

**Comment by franzwilhelm** [comment]
Yup

**Review by thomaslombart — APPROVED** [review]

**Pull request merged**